instrumentctl/g9_driver.py

```diff
@@ -0,0 +1,254 @@
+        assert isinstance(byteString, bytes)
+        return sum(byteString[startByte:endByte + 1]).to_bytes(1, "big") 
+
+    # helper function to convert bytes to bits for checking flags
+    # not currently being used but many be helpful in the future for getting errors
+    def bytes_to_binary(self, byte_string):
+        return ''.join(format(byte, '08b') for byte in byte_string)
+    
+    # this method is made to check the error flags, right not only checks the last 13 bits
+    # of a byte string
+    def check_flags13(self, byteString, norm = 1):
+        assert isinstance(byteString, bytes)
+        # this is for if we only need the last 13 bits (more or less hardcoding this 
+        # just including the rest if it might be helpful in the future
+        if int(self.bytes_to_binary(byteString)[-13:], 2) >= (13 * norm):
+            # all flags we care about are 1
+            return True
+        else:
+            return False
+
+    def response(self):
+        if not self.is_connected():
+            raise ConnectionError("Seiral Port is Not Open.")
+        
+        data = self.ser.read(size=199)
+        self.lastResponse = data
+        if len(data) == 199:
+            alwaysHeader = data[0:3]
+            alwaysFooter = data[-2:]
+            if alwaysHeader != b'\x40\x00\x00' or alwaysFooter != b'\x2A\x0D':
+                raise ValueError("Always bits are incorrect")
+            OCTD = data[7:11]
+            if OCTD != self.msgOptData:
+                raise ValueError("Optional Transmission data doesn't match data sent to the G9SP")
+
+            # Terminal Data Flags
+            SITDF = data[11:17]
+            if not self.check_flags13(SITDF):
+                err = []
+                gates = self.bytes_to_binary(SITDF)
+                for i in range(18):
+                    if gates[-i + 1] == "0":
+                        err.append(i)
+                raise ValueError("An input is either off or throwing an error: {err}")
+            
+            SOTDF = data[17:21]
+            if not self.check_flags13(SOTDF):
+                err = []
+                gates = self.bytes_to_binary(SITDF)
+                for i in range(14):
+                    if gates[-i + 1] == "0":
+                        err.append(i)
+                raise ValueError("There is output(s) off: {err}")
+
+            # Terminal Status Flags
+            SITSF = data[21:27]
+            if not self.check_flags13(SITSF):
+                if self.safety_in_terminal_error(data[31:55]):
+                    raise ValueError("Error was detected in inputs but was not found")
+                
+            SOTSF = data[27:31]
+            if not self.check_flags13(SOTSF):
+                if self.safety_out_terminal_error(data[55:71]):
+                    raise ValueError("Error was detected in outputs but was not found")
+                
+            # Unit status
+            US = data[73:75]
+            if US != 0:
```

Comment:
looks like this attempts to compare a bytes object with an integer. 

should probably be something like `int.from_bytes(US)`